File: clean_leads.py
import csv
import re
import os

def clean_email(email):
    if not email:
        return None
    email = email.strip()
    # Basic email regex
    if re.match(r"[^@]+@[^@]+\.[^@]+", email):
        return email.lower()
    return None

def clean_name(name):
    if not name:
        return "Texas Realtor"
    name = name.strip()
    # Remove some common suffixes/titles if they interfere
    name = re.sub(r",?\s*(Realtor®?|STAR REALTY GROUP|Group|Team).*", "", name, flags=re.IGNORECASE)
    # Handle specific placeholders
    if "Last Name Unknown" in name:
        name = name.split("[")[0].strip()
    if not name:
        return "Texas Realtor"
    return name

def process_leads(input_file, output_file):
    leads = {} # map email -> name
    
    if not os.path.exists(input_file):
        print(f"Error: {input_file} not found.")
        return

    with open(input_file, mode='r', encoding='utf-8', errors='ignore') as f:
        reader = csv.reader(f)
        headers = next(reader) # Skip header line
        
        for row in reader:
            # Block 1: Twitter
            if len(row) > 2:
                name = row[0]
                email = clean_email(row[2])
                if email:
                    leads[email] = clean_name(name)
            
            # Block 2: YouTube
            if len(row) > 13:
                name = row[11]
                email = clean_email(row[13])
                if email and email not in leads:
                    leads[email] = clean_name(name)
            
            # Block 3: Facebook
            if len(row) > 22:
                name = row[20]
                email = clean_email(row[22])
                if email and email not in leads:
                    leads[email] = clean_name(name)

    print(f"Extracted {len(leads)} unique contacts.")

    with open(output_file, mode='w', encoding='utf-8', newline='') as f:
        writer = csv.writer(f)
        writer.writerow(["Name", "Email"])
        for email, name in leads.items():
            writer.writerow([name, email])
    
    print(f"Cleaned leads saved to {output_file}")

if __name__ == "__main__":
    input_path = r"C:\Users\praja\Downloads\x twitter - Sheet1.csv"
    output_path = os.path.join(os.getcwd(), "cleaned_leads.csv")
    process_leads(input_path, output_path)

File: clean_leads.js
const fs = require('fs');
const path = require('path');

function cleanEmail(email) {
    if (!email) return null;
    email = email.trim();
    if (/[^@]+@[^@]+\.[^@]+/.test(email)) {
        return email.toLowerCase();
    }
    return null;
}

function cleanName(name) {
    if (!name || name.trim() === '') return "Texas Realtor";
    name = name.trim();
    // Remove common realtor suffixes
    name = name.replace(/,?\s*(Realtor®?|STAR REALTY GROUP|Group|Team|Licensed Texas Realtor).*/gi, '');
    if (name.includes('[Last Name Unknown]')) {
        name = name.split('[')[0].trim();
    }
    return name || "Texas Realtor";
}

function processLeads(inputPath, outputPath) {
    try {
        const data = fs.readFileSync(inputPath, { encoding: 'utf8', flag: 'r' });
        const lines = data.split(/\r?\n/);
        const leads = new Map(); // email -> name

        // Skip header
        for (let i = 1; i < lines.length; i++) {
            const line = lines[i];
            if (!line.trim()) continue;

            // Simple CSV split (handling basic cases, though nested commas in quotes might be tricky)
            // For this specific sheet, we target:
            // Block 1 (Twitter): Name is Col 1, Email is Col 3
            // Block 2 (YouTube): Name is Col 12, Email is Col 14
            // Block 3 (Facebook): Name is Col 21, Email is Col 23

            // Robust comma split (ignoring commas inside quotes)
            const parts = line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);

            // Block 1: Twitter
            if (parts.length > 2) {
                const name = parts[0].replace(/"/g, '');
                const email = cleanEmail(parts[2]);
                if (email) {
                    leads.set(email, cleanName(name));
                }
            }

            // Block 2: YouTube
            if (parts.length > 13) {
                const name = parts[11].replace(/"/g, '');
                const email = cleanEmail(parts[13]);
                if (email && !leads.has(email)) {
                    leads.set(email, cleanName(name));
                }
            }

            // Block 3: Facebook
            if (parts.length > 22) {
                const name = parts[20].replace(/"/g, '');
                const email = cleanEmail(parts[22]);
                if (email && !leads.has(email)) {
                    leads.set(email, cleanName(name));
                }
            }
        }

        console.log(`Extracted ${leads.size} unique contacts.`);

        const outputData = [['Name', 'Email']];
        for (const [email, name] of leads.entries()) {
            outputData.push([`"${name}"`, email]);
        }

        fs.writeFileSync(outputPath, outputData.map(row => row.join(',')).join('\n'));
        console.log(`Cleaned leads saved to ${outputPath}`);
    } catch (error) {
        console.error("Error processing leads:", error);
    }
}

const inputPath = "C:\\Users\\praja\\Downloads\\x twitter - Sheet1.csv";
const outputPath = path.join(__dirname, "cleaned_leads.csv");

processLeads(inputPath, outputPath);
